tests/unit/test_wide_deep_utils.py

```diff
@@ -40,60 +44,73 @@ def pd_df():
 
 
 @pytest.mark.gpu
-def test_build_feature_columns(pd_df):
-    data, users, items = pd_df
-
-    # Test if wide column has one crossed column
-    wide_columns, _ = build_feature_columns(users, items, model_type='wide')
-    assert len(wide_columns) == 1
-
-    # Test if deep columns have user and item columns
-    _, deep_columns = build_feature_columns(users, items, model_type='deep')
-    assert len(deep_columns) == 2
-
-    # Test if wide and deep columns have correct columns
-    wide_columns, deep_columns = build_feature_columns(users, items, model_type='wide_deep')
-    assert len(wide_columns) == 1
-    assert len(deep_columns) == 2
-
-
-@pytest.mark.gpu
-def test_build_model(pd_df):
+def test_wide_deep(pd_df, tmp_path):
+    """Test `build_feature_columns` and `build_model`"""
     data, users, items = pd_df
 
     # Test wide model
-    wide_columns, deep_columns = build_feature_columns(users, items, model_type='wide')
-    model = build_model('wide_'+MODEL_DIR, wide_columns=wide_columns)
-    assert isinstance(model, tf.estimator.LinearRegressor)
-    model = build_model('wide_'+MODEL_DIR, wide_columns=wide_columns, deep_columns=deep_columns)
+    # Test if wide column has two original features and one crossed feature
+    wide_columns, _ = build_feature_columns(
+        users, items, model_type='wide', crossed_feat_dim=10
+    )
+    assert len(wide_columns) == 3
+    # Check crossed feature dimension
+    assert wide_columns[2].hash_bucket_size == 10
+    # Check model type
+    model = build_model(
+        str(tmp_path / ('wide_' + MODEL_DIR)), wide_columns=wide_columns
```

Comment:
I don't understand what this does?
```str(tmp_path / ('wide_' + MODEL_DIR))```

why is there a division operator here? should it be os.path.join(str(tmp_path), 'wide_' + MODEL_DIR), would this be causing the problem with windows path cleanup?

Reply:
@gramhagen when I use our fixture (tmp) for temporary directory, it causes error (only) on Windows DSVM. So I end up using pytest's fixture tmp_path witch returns `Path` object. And the codes use Path object's syntax `/`.

Reply:
Btw, the error is -- somehow Windows' DSVM holds TF model checkpoints folder and thus TemporaryDirectory cannot cleanup the folder by the time the test has done.

pytest's fixture tmp_path keeps the folder for last three test sessions instead of removing it right away. That's how it avoids the cleanup error. Not a clean solution but at least it is working for now for Windows testing.

Reply:
Ok, I see, I wasn't familiar with that syntax. Do you think there is something that needs to be done to close the tf session or something in order to release the tmp path for cleanup?

Reply:
Maybe the best thing to do is keep it as is and file an issue for using the normal yield fixture with TF

Reply:
I tried `del model` and no luck. Kind of hard to debug because that is happening only from Win-DSVM. @roalexan Any idea about this? FYI no issues when use local Windows machines (both laptop and workstation). Does Win-DSVM handle processes and folder read/write permission differently from local Windows'?

Reply:
Should just be Windows Server 2016 - don't see how it would be different than local. [Link](https://docs.microsoft.com/en-us/azure/machine-learning/data-science-virtual-machine/provision-vm) to product page. Maybe we could use one of these email aliases to ask a question about this: dsvm-product-team@ or Dsvm-interest@.

Reply:
Somewhat related issue found from Tensorflow github issues: https://github.com/tensorflow/tensorflow/issues/20606

I will try to force close tf's FileWriter and see if that solves this issue on Windows DSVM

Reply:
Ok it does solve. I'm working on adding those changes.

Reply:
great, feel free to merge if everything passes